client/src/main/java/io/pravega/client/netty/impl/ConnectionFactoryImpl.java

```diff
@@ -119,6 +120,9 @@ public void initChannel(SocketChannel ch) throws Exception {
                 @Override
                 public void operationComplete(ChannelFuture future) {
                     if (future.isSuccess()) {
+                        Channel ch = future.channel();
```

Comment:
Perhaps rather than blocking here, we could get a future and have result set upon it's completion. This would avoid tieing up the thread

Reply:
Updated the code with a Future based implementation to prevent tieing up the the thread. Please review it.
Also, `future.channel()` is a non-blocking operation as the `ChannelFuture` is already completed. This is used to only for logging purposes.